### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by tjay1760**
# In this Pull request I
Removed current_user method in ApplicationController
[Installed and setup devise]
### Ensured that the app can: 

- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (AbelG101) [inline app/views/devise/sessions/new.html.erb:27]: - [ ] Great job on developing a functional web app! However, it appears that you may have overlooked one of the requirements, which is to ensure that your design matches the provided wireframe. Currently, your web app is utilizing the default styling of the browser, which deviates from the intended design. I recommend updating your design to align with the wireframe specifications.
- Your design:
  firefox_Hus3efygXm
- Wireframe design:
  firefox_F87DeENYem

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @tjay1760 & @HtetWaiYan7191 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Devise is properly setup ✔
- Registration and Sign in work as expected ✔
- Correct use of Git Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/assets/stylesheets/sign.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+input,
+.action {
+  width: 100%;
+  margin: 0;
+  padding: 0;
+  height: 2rem;
+  padding-left: 5px;
+  border-radius: 5px;
+  border: none;
+}
```
Comment (joel-kalema) [inline app/assets/stylesheets/sign.css:32]: - [ ] [OPTIONAL] Good job aligning items, however, I'll suggest making all `input` accessible to facilitate a user to see them easily by removing the `border: none` in your style 👍 

> | | without `border: none` | | with `border: none` | |
> |-|---------|-|--------|-|
> | |Screenshot 2023-10-06 140546| |Screenshot 2023-10-06 140638| |

**Review by joel-kalema — APPROVED** [review]
Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**